crates/bevy_render/src/render_graph/nodes/pass_node.rs

```diff
@@ -216,13 +217,30 @@ where
                         continue;
                     };
 
-                    // get an ordered list of entities visible to the camera
-                    let visible_entities = if let Some(camera_entity) = active_cameras.get(&camera_info.name) {
-                        world.get::<VisibleEntities>(camera_entity).unwrap()
+                    let camera_entity = if let Some(camera_entity) = active_cameras.get(&camera_info.name) {
+                        camera_entity
                     } else {
                         continue;
                     };
 
+                    // get an ordered list of entities visible to the camera
+                    let visible_entities = world.get::<VisibleEntities>(camera_entity).unwrap();
+
+                    // get camera viewport and apply it
+                    if let Ok(viewport) = world.get::<Viewport>(camera_entity) {
+                        let origin = viewport.physical_origin();
+                        let size = viewport.physical_size();
+                        render_pass.set_viewport(
+                            origin.x, origin.y,
+                            size.x, size.y,
+                            // TODO: implement min/max depth
+                            0.0, 1.0,
```

Comment:
Implementing this is actually required for overlapping viewports to work as expected. Otherwise the viewport renders will get mixed up.

Reply:
This change was actually simple enough to include, the only thing missing is the usage example.